templates/onboarding_survey.html

```diff
@@ -0,0 +1,123 @@
+{% extends "layout.html" %}
+{% block header %}
+<script>
+    document.addEventListener('DOMContentLoaded', function () {
+
+        // FIXME: hardercoded index
+        var racecheckboxes = document.querySelectorAll('input[type="checkbox"]');
+        var racenotlisted = racecheckboxes[racecheckboxes.length - 1];
+        racenotlisted.addEventListener('change', function (event) {
+            if(event.target.checked) {
+                var racefields = document.getElementById('racefields');
+                var fielddiv = document.createElement('div');
+                fielddiv.id = 'dynamicracefield';
+                fielddiv.className = 'field';
+                var fieldinput = document.createElement('input');
+                fieldinput.type = 'text';
+                fieldinput.name = 'race';
+                fieldinput.className = 'input';
+                fieldinput.placeholder = 'Please specify';
+                fieldinput.required = 'required';
+                fielddiv.appendChild(fieldinput);
+                racefields.appendChild(fielddiv);
+            } else {
+                var dynamicracefield = document.getElementById('dynamicracefield');
+                if (dynamicracefield) {
+                    dynamicracefield.remove();
+                }
+            }
+        });
+
+    });
+
+    function validateForm() {
+        var checkboxes = document.querySelectorAll('input[type="checkbox"]');
+
+        var allcheckboxes = Array();
+        for (var i = 0; i < checkboxes.length; i++) {
+            console.log(checkboxes[i]);
+            if (checkboxes[i].checked) {
+                allcheckboxes.push(true);
+            }
+        }
+        if (allcheckboxes.length < 1) {
+            return false;
+        }
+    }
+</script>
+{% endblock %}
+{% block content %}
+<div class="column box onboarding-survey">
```

Comment:
note -- form not carefully reviewed. It all looks groovy on a quick pass and I trust your html